### ISSUE TITLE
Simplify route registration methods

### ISSUE THREAD

**Issue by bilus** [bilus/akasha#6]
Do you like the current interface to `CommandRouter`? 

```ruby
  # This is how you link commands to aggregates.
  @command_router.register_default_route(:sign_up, User)

  # Nearly identital to the default handling above but we're setting the admin
  # flag to demo custom command handling.
  @command_router.register_route(:sign_up_admin) do |aggregate_id, **data|
[...]
```

I don't quite like the current method names, what should they be called? Any ideas?

> If/when there's the ultimate DSL for this we can be more nit-picking but I'd like this to be intuitive even now.

**Comment by scoiatael** [comment]
1. What's the exact purpose of having both types? Maybe we should have only one method, which accepts either block or module?
2. Maybe we could simply use a map and avoid DSL at this point? ;)

**Comment by bilus** [comment]
1. I was thinking along the same lines but then kinda decided against. But now that you also bring it up...
2. Like so:

```ruby
@command_router.register_commands(
   sign_up: User,
   sign_out: User,
   sign_up_admin: -> (aggregate_id, **data) {
   }
)
```
?

If so, I like the principle but it is premature to choose the data structure because I kinda feel we may need to have middleware (e.g. for authentication, schema changes) on both levels (global, for entire command router, and for individual commands). On the other hand, it's just lambda so it will compose anyway. :> Damn, you may be right.

@brapacz What do you think? Start with hash?

**Comment by hipek** [comment]
Keep it simple for now so I would vote for hash 👍

**Comment by Holek** [comment]
Would you keep that in the single place, or rather be able to register commands from multiple files?
In particular, how do you want to keep the state of what commands are registered and to where?

**Comment by bilus** [comment]
@Holek In command router.

**Comment by brapacz** [comment]
@bilus +1 for Hash. We can always build something more sophisticated.

**Comment by bilus** [comment]
I started implementing the hash approach but started having second thoughts because I want to add a kind of namespacing for event and command names (and aggregates) so when writing multiple services using the same ES, you don't worry about name clashes. It was a real problem in the NT project (not a huge one but it did exist).

What about having explicit Event and Command classes? Yes? No?

Example:

```ruby
class UserSignedUp < Akasha::Event
  attr_accessor :email, :password

  handled_by UserListMaterializer, :on_user_signed_up
  handled_by Notifier, :on_user_signed_up, async: true
end

class SignUpUser < Akasha::Command
  attr_accessor :email, :password

  handled_by User, :sign_up
end

class SignUpAdmin < Akasha::Command
  attr_accessor :email, :password

  # In the rare case you need a custom handler.
  def handle(aggregate_id)
    user = User.find_or_create(aggregate_id)
    user.sign_up(email: email, password: password, admin: true)
    user.save!
  end
end
```

With the above, we could simplify initialization to just:

```ruby
store = Akasha::Storage::HttpEventStore.new(username: 'admin', password: 'changeit')
Akasha.connect!(store)
```

Namespacing:

```ruby
class SignUpUser < Akasha::Command
  namespace :app_xyz
  attr_accessor :email, :password

  handled_by User, :sign_up
end
```

Routing commands:

```ruby
post '/users/:user_id' do # With CQRS client pass unique aggregate ids.
  SignUpUser.new(params).handle
  'OK'
end
```

@scoiatael @brapacz @hipek @memek? :)

**Comment by bilus** [comment]
Alternative without cleverness (https://github.com/bilus/akasha/pull/10):

```ruby
    # Aggregates will load from and save to in-memory storage.
    repository = Akasha::Repository.new(Akasha::Storage::HttpEventStore.new(username: 'admin', password: 'changeit'), 
                                        namespace: :my_app)
    Akasha::Aggregate.connect!(repository)

    # Set up event listeners.
    event_router = Akasha::EventRouter.new(
      user_signed_up: UserListMaterializer
    )
    event_router.connect!(repository)

    async_event_router = Akasha::AsyncEventRouter.new(
      user_signed_up: Notifier
    )
    async_event_router.connect!(repository) # Returns Thread instance.

    @command_router = Akasha::CommandRouter.new(
      sign_up: User,
      sign_up_admin: ->(aggregate_id, **data) {
        user = User.find_or_create(aggregate_id)
        user.sign_up(email: data[:email], password: data[:password], admin: true)
        user.save!
      }
    )
```

**Comment by bilus** [comment]
Resolved by #10
